### ISSUE TITLE
How to add mega.nz drive ? 

### ISSUE THREAD

**Issue by elig0n** [rclone/rclone#3980]
hi i cannot seem to find mega.nz in the backends list or via config

**Comment by darthShadow** [comment]
https://rclone.org/mega/

**Comment by elig0n** [comment]
Yes I've seen that but it doesn't work. LIke I said there was no "mega" in the list and typing it in didn't help either.

**Comment by darthShadow** [comment]
What version of rclone are you using? Perhaps its too old. I am able to configure and use the Mega backend without issues on the latest rclone beta. It has been available for at least the last couple of stable releases.

**Comment by elig0n** [comment]
rclone v1.50.2
- os/arch: linux/amd64
- go version: go1.13.6

on WSL.
The manual page does show "* Mega"
The backends I see when inside interactive 'rclone config'  are 32 in number and as follows:

```
  1 / 1Fichier
   \ "fichier"
 2 / Alias for an existing remote
   \ "alias"
 3 / Amazon Drive
   \ "amazon cloud drive"
 4 / Amazon S3 Compliant Storage Provider (AWS, Alibaba, Ceph, Digital Ocean, Dreamhost, IBM COS, Minio, etc)
   \ "s3"
 5 / Backblaze B2
   \ "b2"
 6 / Box
   \ "box"
 7 / Cache a remote
   \ "cache"
 8 / Citrix Sharefile
   \ "sharefile"
 9 / Dropbox
   \ "dropbox"
10 / Encrypt/Decrypt a remote
   \ "crypt"
11 / FTP Connection
   \ "ftp"
12 / Google Cloud Storage (this is not Google Drive)
   \ "google cloud storage"
13 / Google Drive
   \ "drive"
14 / Google Photos
   \ "google photos"
15 / Hubic
   \ "hubic"
16 / JottaCloud
   \ "jottacloud"
17 / Koofr
   \ "koofr"
18 / Local Disk
   \ "local"
19 / Mail.ru Cloud
   \ "mailru"
20 / Microsoft Azure Blob Storage
   \ "azureblob"
21 / Microsoft OneDrive
   \ "onedrive"
22 / OpenDrive
   \ "opendrive"
23 / Openstack Swift (Rackspace Cloud Files, Memset Memstore, OVH)
   \ "swift"
24 / Pcloud
   \ "pcloud"
25 / Put.io
   \ "putio"
26 / SSH/SFTP Connection
   \ "sftp"
27 / Transparently chunk/split large files
   \ "chunker"
28 / Union merges the contents of several remotes
   \ "union"
29 / Webdav
   \ "webdav"
30 / Yandex Disk
   \ "yandex"
31 / http Connection
   \ "http"
32 / premiumize.me
   \ "premiumizeme"
```

**Comment by darthShadow** [comment]
Weird, mine shows 36 backends. Just tried a local 1.50.1 beta and its showing up there too.

Would you mind upgrading to the latest version from the official [downloads](https://rclone.org/downloads/) ?

**Comment by ncw** [comment]
> on WSL Debian sid, installed via apt.

I wonder if debian are leaving out the mega backend for some reason. There are no build constraints on the mega backend so it should be present...

Ah yes, I see this in the source package

> debian/patches/0002-Disable-mega-backend.patch

@toddy15 can you tell us the reason for this - It didn't say in the patch. Is there something I could do to make the backend acceptable for Debian? Thanks!

**Comment by soredake** [comment]
https://sources.debian.org/patches/rclone/1.50.2-3/0002-Disable-mega-backend.patch/

**Comment by stratosgear** [comment]
I ended up here, too, since `mega` does not appear in the v1.50.2 version of rclone that is installed on Ubuntu 20.04

**Comment by soredake** [comment]
toddy@debian.org

**Comment by toddy15** [comment]
Hi @ncw,

the reason that the mega backend is disabled is simple: the package "github.com/t3rm1n4l/go-mega" is not available in Debian, therefore I cannot enable that backend.

As soon as t3rm1n4l/go-mega is available as a Debian package, the mega backend can be enabled.

Regards,
Tobias

**Comment by soredake** [comment]
Workaround for Kubuntu 20.10:
```
echo "deb https://packages.azlux.fr/debian/ buster main" | sudo tee /etc/apt/sources.list.d/azlux.list
wget -qO - https://azlux.fr/repo.gpg.key | sudo apt-key add -
```

**Comment by blademckain** [comment]
why for mega don't use [the official Megasync for linux](https://mega.nz/linux/MEGAsync/)?

**Comment by soredake** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=983445

**Comment by darthShadow** [comment]
At this point, we are blocked on go-mega not being available as a debian package (https://github.com/t3rm1n4l/go-mega/issues/27). Posting here will not help resolve this so please focus your efforts on getting go-mega packaged for debian or finding a maintainer who can help with that.

**Comment by maltfield** [comment]
@darthShadow in the meantime, can you please [update the website](https://rclone.org/mega/) with a note indicating that "Mega" may be missing as a remote on some systems, and include a link link to this github issue?

That would have saved me (and certainly main others) a lot of time

**Comment by maltfield** [comment]
For users who prioritize using mega (over using rclone), there's a couple alternatives until this upstream issue is resolved:

1. The official mega MEGAsync GUI client https://mega.io/desktop
2. The official mega MEGAcmd CLI client https://mega.io/cmd
3. The unofficial megatools CLI client https://megatools.megous.com/

Ironically, the first two official packages are *not* in the Debian repos, but there is an RFP for that here

 * https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=939318

But the unofficial megatools package *is* in the Debian repos:

 * https://packages.debian.org/bullseye/megatools

Unfortunately there's an auth bug in Debian 11's version, but the fixed version is available in Debian 12

 * https://github.com/megous/megatools/issues/411

**Comment by firrsefu** [comment]
I've created a "bug" under Work-Needing and Prospective Packages to request go-mega be added to Debian so rclone can work with Mega.nz storage: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1039997

See also https://github.com/t3rm1n4l/go-mega/issues/27

**Comment by Maytha8** [comment]
@darthShadow @ncw I've packaged go-mega in Debian (it just got into Debian Unstable), and I'm now working alongside Tobias to update the rclone package, get Mega re-enabled, among other things.

Feel free to ping me here or email me for anything else. 😄

**Comment by darlezior** [comment]
Hello, Is there any news about this issue? I Just installed rclone INSIDE proxmox and cannot find mega, thanks

**Comment by jbonadiman** [comment]
@Maytha8 any updates on this? I tried to check the debian feeds, but I understand very little of the process.

**Comment by Maytha8** [comment]
@jbonadiman There are currently a bunch of other obstacles to updating rclone, mainly a big Google Golang package transition, that involves google.golang.org/grpc (golang-google-grpc package) and a large swathe of stuff that depends on it and needs to be patched/updated to build with new versions of the Google packages.

**Comment by kamzata** [comment]
Any news on this?

**Comment by richardfrangie** [comment]
> Any news on this for Debian 12 bookworm?

I had the same problem. I had installed the package manager version `1.60.1`, but then I installed the new version available `1.68.1` from the [rclone main page](https://rclone.org/install/#linux). I installed it in `~/.local/bin`, to have it separate, and now it recognizes `mega` in the `rclone config` options.

**Comment by kamzata** [comment]
> > Any news on this for Debian 12 bookworm?
> 
> I had the same problem. I had installed the package manager version `1.60.1`, but then I installed the new version available `1.68.1` from the [rclone main page](https://rclone.org/install/#linux). I installed it in `~/.local/bin`, to have it separate, and now it recognizes `mega` in the `rclone config` options.

It seems to work! Thanks

**Comment by uluckas** [comment]
It seems golang-github-t3rm1n4l-go-mega [has arrived in testing](https://tracker.debian.org/pkg/golang-github-t3rm1n4l-go-mega)

**Comment by maltfield** [comment]
Has anyone tested if the mega backend is working with the rclone package that's installed from the default Debian repos in Debian 13?

Debian 13 (trixie) was just released last month.

 * https://bits.debian.org/2025/08/trixie-released.html

**Comment by TulossSolutions** [comment]
On **Ubuntu 24.04.3**, when installing using apt, Mega is still missing from the list. So I had to use the install script from rclone website https://rclone.org/downloads/ :

`sudo -v ; curl https://rclone.org/install.sh | sudo bash`

**Comment by maltfield** [comment]
Unfortunately, this still appears to be missing in Debian 13 :(

Can someone say what's blocking this from being fixed, so it can hopefully be done in Debian 14?
